docs/user_guide.md

```diff
@@ -160,12 +167,25 @@ log arguments:
 analyzer arguments:
 
   -j JOBS, --jobs JOBS
+  -c, --clean
+  -n, --name
```

Comment:
Why do we have a `--name` here? `check` does not store anything.

Reply:
Maybe we don't need this. I inserted this argument here because `analyze` has a `name` flag too and this is just a wrapper script which runs `analyze` among others. By the way `analyze` uses `name` argument to write this name to `metadata.json` which I think is not used anywhere. We could remove `name` from `analyze` too, but that might break some scripts which use this if any.

Reply:
If a `--name` is given to `analyze`, it will be written into the `metadata.json`. It **is** used. `store` runs over the given report directories, eating up the `metadata.json`s therein, and construct the name for the stored analysis automatically.

https://github.com/Ericsson/codechecker/blob/a096bccae947f8168fceb4ce108efba8354d1ba6/libcodechecker/libhandlers/store.py#L273-L282

Technically, agreeing with @gyorb: `check` does no longer wrap `store`, so for a full-fledged `check` execution, `--name` is not needed.

Version 6 was rolled **without** `--name` in check, so there is no need to introduce it. (The only reason would be "preparing" an output_dir which contains an automatically `store`-able structure.) Old scripts which use `-n/--name` for `check` are using v5 syntax, which means that they **should** break, as that invocation is not compatible with the current behaviour.

Reply:
@bruntib @whisperity I've checked and the analyze  and it has the `--name` argument too, which will write the given name (run name) into the metadata so at the store you do not have to give a value for it.
Based on the help message of the store command if a name is detected in the `metadata.json` it will be used. Does this work still this way?

Reply:
Yes, `analyze` and `store` works that way. (I posted the revelant code snipped from `store` above.) But that doesn't imply `check` needs to have a `--name` argument. Due to it not having a use in a full `check` invocation, I'd rather not have it wrapped.